### PR TITLE
[DRAFT] More emoticon variety, custom emote wheel.

### PR DESCRIPTION
**Pull request by VoxelDoesCode**
Direct link to #5708 . I want a way to have the emotes YOU want to have, as well as some newer ones to fit your need. These new emotes are:
HELP, WAIT, SPEC, ACHE, WIFI, CLIP, YES, NO, TALK

image

Emoticons file used in testing (22/11/22 - not embedded):

emoticons_extend

These new images are not final, but they give out a very clean PoC for what they intend to be in the release version.

## TODO
- [x] Let new emoticons show in server
- [ ] Backwards compatibility (allowing both old and new versions of emote files)
- [ ] Sprite cleanup
- [ ] Ability to change emotes in wheel
- [ ] Menu UI/UX

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/gameclient.h [hunk 1]
```diff
@@ -645,7 +645,7 @@ class CGameClient : public IGameClient
 
 	struct SClientEmoticonsSkin
 	{
-		IGraphics::CTextureHandle m_aSpriteEmoticons[16];
+		IGraphics::CTextureHandle m_aSpriteEmoticons[25];
 	};
```
Comment (fokkonaut) [inline src/game/client/gameclient.h:649]: NUM_EMOTICONS

Comment (VoxelDoesCode) [reply src/game/client/gameclient.h:649]: Fixed!

**Comment by VoxelDoesCode** [comment]
Might change TALK, since it implies more about the console than talking. Don't know what to make it into.

**Comment by heinrich5991** [comment]
HELP is one I'd not like, it's language specific. I realize we already have those, but we don't need to make it worse.

What is the crossed out WiFi symbol supposed to mean?

**Comment by VoxelDoesCode** [comment]
It's meant to indicate lag, or high ping. Ravie said you could just type in lag in chat, but I think this is quicker.

**Comment by VoxelDoesCode** [comment]
Might just cut it down to just 4 new emotes. ACHE, YES, NO and WAIT.

**Comment by Vy0x2** [comment]
I liked all of the emoticons you used in 22/11/22 and would love to see you continue working on it :-)

**Comment by def-** [comment]
Custom emotes will be invalid anyway because they will miss some graphics. So just rename the file and increase the size as you see fit.

**Comment by heinrich5991** [comment]
> Custom emotes will be invalid anyway because they will miss some graphics. So just rename the file and increase the size as you see fit.

I'd like to keep compatibility with the old emote images, there are soo many of them. I'd use the default images for the missing ones.

**Comment by VoxelDoesCode** [comment]
I'll close this for now, since this is not a priority for me currently. And, I don't know how to make image sizes flexible with one type of element.

**Comment by fokkonaut** [comment]
@VoxelDoesCode sorry, I missclicked.

**Pull request closed without merging**